test/MarginAccount.test.ts

```diff
@@ -0,0 +1,486 @@
+      //   await marginAccountTester.testRemoveShort(otoken.address, 1, 1)
+      //   const vault = await marginAccountTester.getVault()
+      //   assert.equal(1,2)
+      // })
+      // it('should revert if trying on the wrong vault', async () => {
+      //   // await expectRevert(marginAccountTester.testAddShort(otoken.address, 10, 5, {from: deployer}), 'revert')
+      //   // await expectRevert(await marginAccountTester.testAddShort(Otoken, 10, 3, {from: deployer}), 'revert')
+      //   // await expectRevert(await marginAccountTester.testAddShort(Otoken, 10, 100, {from: deployer}), 'revert')
+      // })
+    })
+
+    describe('Clear vault', () => {
+      before('redeploying', async () => {
+        marginAccountTester = await MarginAccountTester.new()
+      })
+
+      it('vaultIds should no longer be zero, owner should be null', async () => {
+        account = await marginAccountTester.getAccount({from: deployer})
+        assert.equal(account.owner, testAccount.owner, 'MarginAccount.Account owner addr mismatch')
+      })
+
+      it('should add weth collateral', async () => {
+        await marginAccountTester.testAddCollateral(weth.address, 10, 0)
+      })
+
+      it('should add long oTokens', async () => {
+        await marginAccountTester.testAddLong(otoken.address, 10, 0)
+      })
+
+      it('should add short oTokens', async () => {
+        await marginAccountTester.testAddShort(otoken2.address, 10, 0)
+      })
+
+      it('ensure that the vault is not empty before clearing', async () => {
+        const vault = await marginAccountTester.getVault()
+        assert.notEqual(vault.shortAmounts.length, 0, 'shortAmounts.length should not be 0')
+        assert.notEqual(vault.longAmounts.length, 0, 'longAmounts.length should not be 0')
+        assert.notEqual(vault.collateralAmounts.length, 0, 'collateralAmounts.length should not be 0')
+        assert.notEqual(vault.shortOtokens.length, 0, 'shortOtokens.length should not be 0')
+        assert.notEqual(vault.longOtokens.length, 0, 'longOtokens.length should not be 0')
+        //TODO: change the following
+        assert.notEqual(vault.collateralAssets.length, 0, 'collateralAssets.length should not be 0')
+      })
+
+      it('ensure that the vault is empty after clearing', async () => {
+        const vault = await marginAccountTester.getVault()
+        await marginAccountTester.testClearVault()
+        assert.equal(vault.shortAmounts.length, 0, 'shortAmounts.length should be 0')
+        assert.equal(vault.longAmounts.length, 0, 'longAmounts.length should be 0')
+        assert.equal(vault.collateralAmounts.length, 0, 'collateralAmounts.length should be 0')
+        assert.equal(vault.shortOtokens.length, 0, 'shortOtokens.length should be 0')
+        assert.equal(vault.longOtokens.length, 0, 'longOtokens.length should be 0')
+        assert.equal(vault.collateralAssets.length, 0, 'collateralAssets.length should be 0')
+      })
+    })
```

Comment:
Can we do some tests that arbitrarily test combinations of add, remove, clear?

Reply:
that's integration tests no? thot that comes later unless this would also be part of unit testing

Reply:
yeah IMO those are integration tests

Reply:
well that is reasonable to test for this file imo.